### ISSUE TITLE
k8s.service configuration doesn't seem to work using installer

### ISSUE THREAD

**Issue by tbarrella** [istio/istio#19517]
Sorry if I'm missing something here. In the meantime using Helm values works, but just reporting this. Also, maybe slightly related to #15245, in that it feels like it would be easier to be sure that the right thing is being modified if it were possible to use something familiar like Kustomize

**Bug description**
Using `k8s.service` when modifying a component doesn't seem to change anything

**Expected behavior**
Can adjust the Service associated with a component using the `service` part of a `KubernetesResourcesSpec`

**Steps to reproduce the bug**
- Create IstioControlPlane config `control-plane.yaml` like
```yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  gateways:
    components:
      ingressGateway:
        k8s:
          service:
            selector:
              doesnt-seem-to: work
            type: NodePort
```
- `istioctl manifest generate -f control-plane.yaml | less`
- View the spec for the istio-ingressgateway Service, which is the same as the default
- Also, adding something like `externalIPs: ["1.2.3.4"]` causes an error
>input file failed validation with the following errors: validate path Gateways.Components.IngressGateway.K8S.Service, value: 1.2.3.4, expected ptr, got string
- But, overriding the Deployment strategy with `strategy` does work

**Version**
1.4.0

**How was Istio installed?**
istioctl

**Comment by howardjohn** [comment]
I think this is fixed by #20233

**Comment by richardwxn** [comment]
yes the service part configuration would be fixed by above PR, which was merged in master already
